### ISSUE TITLE
go get failing with a "SECURITY ERROR".

### ISSUE THREAD

**Issue by bfitzsimmons** [newrelic/go-agent#103]
I'm currently receiving this error when installing the agent. It's probably not likely that this is an actual man-in-the-middle attack, so I thought I would mention it.

```shell
$ go get github.com/newrelic/go-agent@v2.12.0 
go: finding github.com/newrelic/go-agent v2.12.0
go: downloading github.com/newrelic/go-agent v2.12.0+incompatible
verifying github.com/newrelic/go-agent@v2.12.0+incompatible: checksum mismatch
        downloaded: h1:Cb7DO0unyxvi3ZpXd1BwfFTdGGSdcg8mSU5HKuN+xKU=
        sum.golang.org: h1:qh9AHY4YR9Tpq2beQ7Nik51Mxqw8XknzBLQ2F11Vfe0=

SECURITY ERROR
This download does NOT match the one reported by the checksum server.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```

**Comment by newrelic-eheinlein** [comment]
Thank you for reporting this!  We will look into figuring out why this is happening.

**Comment by newrelic-eheinlein** [comment]
@bfitzsimmons My initial testing isn't able to reproduce this; I'm continuing to investigate, but would like to check: what version of Go are you running?

**Comment by bfitzsimmons** [comment]
@newrelic-eheinlein Thanks for looking into it. I'm currently running Go 1.13.

```# go version
go version go1.13 darwin/amd64
```

**Comment by bfitzsimmons** [comment]
Aaaaaand...after seeing the error most of the day, it's no longer occurring. I'm wondering if the new Go proxy has some bugs that need to be shaken out.

**Comment by bfitzsimmons** [comment]
I'm definitely getting the new version -- the interfaces I generated for `newrelic.Transaction` are broken again.

**Comment by willnewrelic** [comment]
@bfitzsimmons   Thanks for the updates!

We are learning about Go modules and the checksum system:  We are happy that the problem has disappeared!

Also, sorry about the `newrelic.Transaction` breakage:  We should have released the transaction as a pointer to a struct rather than an interface for easier upgrades.  We made pursue this change in a major version release in the future. Stay tuned!

I'm going to close this issue.  Don't hesitate to reopen it or get in touch if something else needs attention!

**Comment by niels-s** [comment]
I'm surprised to hear the error disappeared 4 days back for you guys. Today I'm still experiencing the same issue although the checksums I receive don't match the ones originally reported:

```
go: downloading github.com/newrelic/go-agent v2.12.0+incompatible
verifying github.com/newrelic/go-agent@v2.12.0+incompatible: checksum mismatch
	downloaded: h1:qh9AHY4YR9Tpq2beQ7Nik51Mxqw8XknzBLQ2F11Vfe0=
	go.sum:     h1:Cb7DO0unyxvi3ZpXd1BwfFTdGGSdcg8mSU5HKuN+xKU=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.
```

Any further ideas on how to solve this? Do you know what the final correct checksum is?

**Comment by willnewrelic** [comment]
Hey @bfitzsimmons and @niels-s

We think the likely source of this problem is that we moved the v2.12.0 tag from one commit to another to add a commit to the release to fix mistakes in the README.  This was a mistake we regret.  Sorry for the bother!

Fortunately, we will do a new release (v2.13.0) very soon (probably today!), and so we are hopeful this issue will disappear completely.

**Comment by willnewrelic** [comment]
Hi @bfitzsimmons and @niels-s

We have just released version 2.13.0.  I'm hopeful this helps resolve issues!

**Comment by niels-s** [comment]
Thanks updating to v2.13.0 solved the issue 🎉
